server/src/main/java/org/elasticsearch/common/xcontent/XContentHelper.java

```diff
@@ -383,4 +383,58 @@ public static XContentType xContentType(BytesReference bytes) {
         BytesRef br = bytes.toBytesRef();
         return XContentFactory.xContentType(br.bytes, br.offset, br.length);
     }
+
+    /**
+     * Returns the contents of an object as an unparsed BytesReference
+     *
+     * This is useful for things like mappings where we're copying bytes around but don't
+     * actually need to parse their contents, and so avoids building large maps of maps
+     * unnecessarily
+     */
+    public static BytesReference childBytes(XContentParser parser) throws IOException {
```

Comment:
This is in XContentHelper rather than directly on XContentParser because we want to use BytesReference here, which is in `server` due to its lucene dependency :(